src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -2,52 +2,70 @@
 
-import java.util.List;
+import java.util.HashSet;
 
+import javafx.collections.ObservableList;
 import seedu.address.commons.core.Messages;
-import seedu.address.commons.core.index.Index;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
 import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+
 
 /**
  * Deletes a person identified using it's displayed index from the address book.
  */
 public class DeleteCommand extends Command {
 
-    public static final String COMMAND_WORD = "delete";
+    public static final String COMMAND_WORD = "deletefriend";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the person identified by the index number used in the displayed person list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
+            + ": Deletes the friend identified by name.\n"
+            + "Parameters: n/NAME (must be name of a friend who exists in Amigos)\n"
+            + "Example: " + COMMAND_WORD + " n/John Doe";
 
-    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
+    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted friend: %1$s";
 
-    private final Index targetIndex;
+    private final Name nameOfPersonToDelete;
 
-    public DeleteCommand(Index targetIndex) {
-        this.targetIndex = targetIndex;
+    public DeleteCommand(Name nameOfPersonToDelete) {
+        this.nameOfPersonToDelete = nameOfPersonToDelete;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Person> lastShownList = model.getFilteredPersonList();
+        //List<Person> lastShownList = model.getFilteredPersonList();
+
+        Person personWithNameToDelete = new Person(nameOfPersonToDelete, new Phone("12345678"),
+                new Email("dummyemail@gmail.com"), new Address("Dummy address"), new HashSet<>());
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        if (!model.hasPerson(personWithNameToDelete)) { //model.hasPerson considers 2 Persons with same name
+            //to be the same Person
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_NAME);
         }
 
-        Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
-        model.deletePerson(personToDelete);
-        return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, personToDelete));
+        ObservableList<Person> personList = model.getAddressBook().getPersonList();
+
+        final Person[] personToDelete = new Person[1];
```

Comment:
I wrapped the personToDelete in a final 1-size array so that forEach can be used to iterate through the personList as forEach only works on values that are effectively final!